File: output_CodeWhisperer/python/Easy/106-base-7.py
#https://leetcode.com/problems/base-7/
#Given an integer num, return a string of its base 7 representation.
# 
#Example 1:
#Input: num = 100
#Output: "202"
#Example 2:
#Input: num = -7
#Output: "-10"
# 
#Constraints:
#	-107 <= num <= 107
class Solution(object):
    def convertToBase7(self, num):
        """
        :type num: int
        :rtype: str
        """
        if num == 0:
            return "0"
        negative = num < 0
        num = abs(num)
        res = ""
        while num:
            res = str(num % 7) + res
            num //= 7
        if negative:
            res = "-" + res
        return res
        

File: output_CodeWhisperer/cpp/Easy/106-base-7.cpp
/**https://leetcode.com/problems/base-7/ */
//Given an integer num, return a string of its base 7 representation.
// 
//Example 1:
//Input: num = 100
//Output: "202"
//Example 2:
//Input: num = -7
//Output: "-10"
// 
//Constraints:
//	-107 <= num <= 107
class Solution {
public:
    string convertToBase7(int num) {
        if(num == 0){
            return "0";
        }
        bool isNegative = num < 0;
        num = abs(num);
        string result = "";
        while(num > 0){
            result = to_string(num % 7) + result;
            num /= 7;
        }
        if(isNegative){
            result = "-" + result;
        }
        return result;    
    }
};